frontend/src/components/pages/RegistrantExperience/PersonalInfo/personalInfoReducer.tsx

```diff
@@ -0,0 +1,166 @@
+  >,
+  action: PersonalInfoReducerDispatch,
+): void => {
+  setCampers((campers: RegistrantExperienceCamper[]) => {
+    const newCampers: RegistrantExperienceCamper[] = JSON.parse(
+      JSON.stringify(campers),
+    ); // Deep Copy
+    switch (action.type) {
+      case PersonalInfoActions.ADD_CAMPER: {
+        const { campSessions } = action as AddCamper;
+        // Add a unique camper entry for all registered camp sessions
+        /* eslint-disable-next-line */
+        const campSessionIds = Array.from(campSessions.map((campSession) => campSession.id));
+        newCampers.push({
+          id: "",
+          campSessions: campSessionIds,
+          firstName: "",
+          lastName: "",
+          age: NaN,
+          contacts: [],
+          registrationDate: new Date(),
+          hasPaid: false,
+          chargeId: "",
+          charges: {
+            camp: NaN,
+            earlyDropoff: NaN,
+            latePickup: NaN,
+          },
+          optionalClauses: [],
+        });
+
+        // inject contact info
+        /* eslint-disable-next-line */
+            for (const contact of newCampers[0].contacts){
+          newCampers[newCampers.length - 1].contacts.push(
+            JSON.parse(JSON.stringify(contact)),
+          ); // Deep copy the contact
+        }
+        break;
+      }
+      case PersonalInfoActions.DELETE_CAMPER: {
+        const { camperId } = action as DeleteCamper;
+        newCampers.splice(camperId, 1);
+        break;
+      }
+      case PersonalInfoActions.UPDATE_CAMPER: {
+        const { camperId, field, data } = action as UpdateCamper;
+        if (field === "firstName") newCampers[camperId][field] = data as string;
+        else if (field === "lastName")
+          newCampers[camperId][field] = data as string;
+        else if (field === "age") newCampers[camperId][field] = data as number;
+        else if (field === "allergies")
+          newCampers[camperId][field] = data as string;
+        else if (field === "specialNeeds")
+          newCampers[camperId][field] = data as string;
+        break;
+      }
+      case PersonalInfoActions.UPDATE_CONTACT: {
+        const { contactId, field, data } = action as UpdateContact;
+        /* eslint-disable-next-line */
+        for (const camper of newCampers) {
+          if (field === "firstName")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "lastName")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "email")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "phoneNumber")
+            camper.contacts[contactId][field] = data as string;
+          else if (field === "relationshipToCamper")
+            camper.contacts[contactId][field] = data as string;
+        }
+        break;
+      }
+      default:
+    }
+    return newCampers;
+  });
+};
+
+export const usePersonalInfoHook = (
```

Comment:
Could we rename this to `usePersonalInfoDispatcher` or something like that which you find appropriate? Hook seems very broad and can be confusing as to what the hook actually does

Reply:
Yep! Let me change it.